File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        size = len(arr)
        res = [0] * size
        res[size - 1] = -1
        max_num = arr[size - 1]

        for i in range(size - 1, 0, -1):
            max_num = max(arr[i], max_num)
            res[i - 1] = max_num
            
        return res
        

File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int lf = 0; int rt = heights.length - 1;
        int res = 0, max = 0;

        while(lf < rt){
            int dist = rt - lf;
            max = Math.min(heights[lf], heights[rt]) * dist;
            if(max > res){
                res = max;
            }

            if(heights[lf] < heights[rt]){
                lf++;
            } else{
                rt--;
            }
        }

        return res;
    }
}